src/Admin/fields/LockedTextInput/index.tsx

```diff
@@ -0,0 +1,73 @@
+import Notice from '@givewp/admin/components/Notices';
+import {AdminSectionField} from '@givewp/components/AdminDetailsPage/AdminSection';
+import {useEffect, useState} from '@wordpress/element';
+import {__} from '@wordpress/i18n';
+import {useFormContext, useFormState} from 'react-hook-form';
+import {EditIcon} from './Icons';
+import styles from './styles.module.scss';
+
+interface LockedTextInputProps {
+    name: string;
+    label: string;
+    description: string;
+    placeholder?: string;
+    warningMessage: string;
+}
+
+/**
+ * @unreleased
+ */
+export default function LockedTextInput({
+    name,
+    label,
+    description,
+    placeholder,
+    warningMessage
+}: LockedTextInputProps) {
+    const {register} = useFormContext();
+    const {errors, isSubmitSuccessful} = useFormState();
+    const [isEditing, setIsEditing] = useState(false);
+
+    const handleEditClick = () => {
+        setIsEditing(!isEditing);
+    };
+
+    // Reset editing mode when form is successfully submitted
+    useEffect(() => {
+        if (isSubmitSuccessful) {
+            setIsEditing(false);
+        }
+    }, [isSubmitSuccessful]);
+
+    return (
+        <AdminSectionField error={errors[name]?.message as string}>
+            <label htmlFor={name}>{label}</label>
+            <p className={styles.description}>
+                {description}
+            </p>
+            <div className={styles.inputContainer}>
+                <input
+                    id={name}
+                    type="text"
+                    className={`givewp-admin-field-input ${styles.input}`}
```

Comment:
Don't we input have styles coming from the parent Admin screen component? Also, I don't think we're using classnames like this anywhere else.